### ISSUE TITLE
Cuda zeros output values

### ISSUE THREAD

**Issue by schmohlo** [facebookresearch/SparseConvNet#41]
Hello,

When i run the examples on cpu, everything works fine.
When i use cuda however, training is not making any progress (loss stays roughly the same).

I figured out, that on cuda modules like SubmanifoldConvolution and BatchNormReLU return only zeros.
[scn_cuda_bug_demo.py](https://github.com/facebookresearch/SparseConvNet/files/2119926/scn_cuda_bug_demo.txt)

Any Ideas?

I'm using python 3.5, cuda9.0 and pytorch0.4.0.
Plain PyTorch works fine both on cpu and cuda.

**Comment by btgraham** [comment]
Thanks for the bug report. I have just updated SparseConvNet (to deal with changes in PyTorch's backend), so please 'git pull' and 'bash build.sh'

What, please, is the output from running:
```
import torch, sparseconvnet as scn
conv = scn.SubmanifoldConvolution(2, 1, 1, 3, False).cuda()
print(conv.weight)
input=scn.InputBatch(2, 5)
input.add_sample()
input.set_location(torch.LongTensor([1, 1]), torch.FloatTensor([1]), 0)
input.set_location(torch.LongTensor([2, 3]), torch.FloatTensor([1]), 0)
input.set_location(torch.LongTensor([3, 2]), torch.FloatTensor([1]), 0)
input.cuda()
print(conv(input))
```

**Comment by schmohlo** [comment]
Thank you for your quick reply.

I have installed the update, and now everything seems to work fine!
The output is:
```
Parameter containing:
tensor([[[ 0.1646]],
        [[-0.1343]],
        [[ 0.4031]],
        [[ 0.1469]],
        [[ 0.2419]],
        [[-0.0212]],
        [[-0.7108]],
        [[-0.5201]],
        [[ 0.3016]]], device='cuda:0')
SparseConvNetTensor<<features=tensor([[ 0.2419],
        [-0.4689],
        [ 0.6450]], device='cuda:0')coordinates=tensor([[ 1,  1,  0],
        [ 2,  3,  0],
        [ 3,  2,  0]])spatial size=tensor([ 5,  5])>>
```
Unfortunately I can't reproduce the error by reinstalling the older commit.

Thank you very much indeed.
